app/Classes/Controller/MonksyncController.php

```diff
@@ -0,0 +1,576 @@
+						if(self::insertLdapInMonk($itL) != true) break;
+						
+						$itL++;
+					} else if ($sdiff < 0) {
+						//In LDAP nicht vorhanden
+						self::trace(9,__LINE__, "In LDAP nicht vorhanden monkIds=".$this->monkIds[$itM]);
+						$midx=self::searchMonkById(intval($this->monkIds[$itM]));
+						self::trace(9,__LINE__, "In LDAP nicht vorhanden midx=".$midx);
+						self::trace_r(9,__LINE__, "In LDAP nicht vorhanden monkUsers=",$this->monkUsers[$midx]);
+						self::trace(9,__LINE__, "In LDAP nicht vorhanden ".$this->ldapIds[$itL].  " ".$this->monkIds[$itM]);
+						if($midx>=0 && !self::isMarkedInvalid($this->monkUsers[$midx]['email']))  {
+							$this->monkUsers[$midx]['email'].=".invalid";
+							if(self::updateMonkData($midx)===true) $this->stats['setFlagInMonk']++;
+							}
+						$itM++;
+					} else { /* $sdiff==0 */
+						//In beiden vorhanden
+						self::trace(9,__LINE__, "show gleich ".$this->ldapIds[$itL].  " ".$this->monkIds[$itM]);
+						$midx=self::searchMonkById(intval($this->ldapIds[$itL]));
+						$lidx=self::searchLdapById(intval($this->monkIds[$itM]));
+						if ($midx<0 || $lidx<0) {
+							self::error(__LINE__,
+									"Ldap-Daten nicht gefunden (".$this->monkIds[$itL].")");
+						}
+						//Mailadresse in Monk und in Ldap unterschiedlich
+						if (strcasecmp(self::stripInvalid($this->monkUsers[$midx]['email'])
+									,self::stripInvalid($this->ldapUsers[$lidx]['email']))) {
+							self::trace(9,__LINE__, "show midx=".$midx." lidx=".$lidx);
+							self::trace(9,__LINE__, "show midx_email=".$this->monkUsers[$midx]['email']);
+							self::trace(9,__LINE__, "show this->ldapUsers[$lidx]=".$this->ldapUsers[$lidx]['email']);
+
+							$this->monkUsers[$midx]['email'] =$this->ldapUsers[$lidx]['email'];
+							self::trace_r(9,__LINE__, "show this->ldapUsers[$lidx]=",$this->ldapUsers[$lidx]);
+							self::trace_r(9,__LINE__, "show monkUsers=",$this->monkUsers[$midx]);
+							self::updateMonkData($midx);
+						////Mailadresse gleich aber Valid-Kenzeichen unterschiedlich
+						} else if ( self::isMarkedInvalid($this->monkUsers[$midx]['email'])
+								!=  self::isMarkedInvalid($this->ldapUsers[$lidx]['email'])) {
+							self::trace(9,__LINE__, "show midx=".$midx." lidx=".$lidx);
+							self::trace(9,__LINE__, "show midx_email=".$this->monkUsers[$midx]['email']);
+							self::trace(9,__LINE__, "show this->ldapUsers[$lidx]=".$this->ldapUsers[$lidx]['email']);
+
+							//In Ldap invalid aber in Monk valid --> An monk uebertragen
+							if(self::isMarkedInvalid($this->ldapUsers[$lidx]['email'])==true) {
+								self::trace(9,__LINE__, "show an Monk midx=".$midx." lidx=".$lidx);
+								$this->monkUsers[$midx]['email'] =$this->ldapUsers[$lidx]['email'];
+								self::updateMonkData($midx);
+								//In Ldap valid aber in Monk invalid --> An ldap uebertragen
+							} else if(self::isMarkedInvalid($this->monkUsers[$midx]['email'])==true) {
```

Comment:
".invalid" ist bei uns die in LDAP die Markierung für ungültige Adressen. Das taucht also immer in LDAP zuerst auf, nie in ListMonk.
Oder markiert ListMonk ungülitge Adressen auf dieselbe Weise als ungültig?
Gibt es einen Mechanismus, um eine als ungültig geflaggte E-Mail-Adresse jemals wieder freizuschalten bei ListMonk?